src/net/sourceforge/plantuml/FileFormat.java

```diff
@@ -224,25 +224,47 @@ public boolean matchesProperty(String propertyName) {
 
 	private StringBounder getTikzStringBounder(final TikzFontDistortion tikzFontDistortion) {
 		return new StringBounderRaw(FileFormat.gg.getFontRenderContext()) {
+
+			private final LatexManager latexManager = new LatexManager(tikzFontDistortion.getTexSystem(), tikzFontDistortion.getTexPreamble());
+
 			public String toString() {
 				return "FileFormat::getTikzStringBounder";
 			}
 
 			protected XDimension2D calculateDimensionInternal(UFont font, String text) {
-				text = text.replace("\t", "    ");
-				final XDimension2D w1 = getJavaDimension(font.goTikz(-1), text);
-				final XDimension2D w2 = getJavaDimension(font.goTikz(0), text);
-				final XDimension2D w3 = getJavaDimension(font.goTikz(1), text);
-				final double factor = (w3.getWidth() - w1.getWidth()) / w2.getWidth();
-				final double distortion = tikzFontDistortion.getDistortion();
-				final double magnify = tikzFontDistortion.getMagnify();
-				final double delta = (w2.getWidth() - w1.getWidth()) * factor * distortion;
-				return w2.withWidth(Math.max(w1.getWidth(), magnify * w2.getWidth() - delta));
+				double[] widthHeightDepth = latexManager.getWidthHeightDepth(styleText(font, text));
+				return new XDimension2D(widthHeightDepth[0], widthHeightDepth[1] + widthHeightDepth[2]);
 			}
 
 			public boolean matchesProperty(String propertyName) {
 				return false;
 			}
+
+			public double getDescent(UFont font, String text) {
+				double[] widthHeightDepth = latexManager.getWidthHeightDepth(styleText(font, text));
+				return widthHeightDepth[2];
+			}
+
+			protected String styleText(UFont font, String text) {
+				StringBuilder sb = new StringBuilder();
+				final boolean italic = font.isItalic();
+				final boolean bold = font.isBold();
+
+				if (italic)
+					sb.append("\\textit{");
+
+				if (bold)
+					sb.append("\\textbf{");
+
+				sb.append(text);
```

Comment:
The tex source need to synced with:
https://github.com/plantuml/plantuml/blob/83a4bfbdb398ee2e15657c77e80d142321b89d16/src/net/sourceforge/plantuml/tikz/TikzGraphics.java#L386-L403

However, to capture box, LatexManager use catcode to avoid protect.

Reply:
Thanks for your contribution, but I am not sure to understand your point here.
Could you explain a little?

Reply:
Currently, plantuml use Java to calculate dimension then render them with LaTeX, so they don't match.

Codes to calculate dimension for text (master version):
https://github.com/plantuml/plantuml/blob/83a4bfbdb398ee2e15657c77e80d142321b89d16/src/net/sourceforge/plantuml/FileFormat.java#L231-L241

Codes to display text:
https://github.com/plantuml/plantuml/blob/83a4bfbdb398ee2e15657c77e80d142321b89d16/src/net/sourceforge/plantuml/tikz/TikzGraphics.java#L317-L353

As they are unmatched, so it would show weird results.

This PR use LaTeX to calculate dimension for text:
https://github.com/plantuml/plantuml/blob/05112b7bd5ab5c9bbbc7aa43639d9f602e15a3ee/src/net/sourceforge/plantuml/FileFormat.java#L234-L237

However, the LaTeX source to calculate dimension may be slight different from the LaTeX source to render text, as there are `protectedText` in the LaTeX source when redner:

https://github.com/plantuml/plantuml/blob/83a4bfbdb398ee2e15657c77e80d142321b89d16/src/net/sourceforge/plantuml/tikz/TikzGraphics.java#L386-L403

Furthermore, there maybe some other problem. For example in https://github.com/plantuml/plantuml/issues/1606, the text is rendered by every word.

Weird results:
<img width="190" alt="weird" src="https://github.com/user-attachments/assets/c81c163d-59bd-4f34-bdf6-4262712eda09">

<img width="746" alt="weird" src="https://github.com/user-attachments/assets/f1e1a5f7-e4ed-40b5-8dc0-8e5df602bea6">

However, as the API returns the `height` and `depth`, it could be fixed if `text height` is specified with the maximum of the entire words, and `text depth` is re-calculated with the actual `text height` and `text depth` of single word (ref https://tikz.dev/tikz-shapes#sec-17.4.4)